### ISSUE TITLE
Deployment to an environment should not be allowed if a pending deployment is waiting on manual intervention for the same project

### ISSUE THREAD

**Issue by shaunmarx** [OctopusDeploy/Issues#4564]
Currently, we allow a deployment to an environment for a specific project even when there is a pending deployment which is waiting for manual intervention. This can cause some problems when dealing using terraform plan and apply with a manual intervention step. 

Source: https://secure.helpscout.net/conversation/581656122?folderId=557077

**Comment by MJRichardson** [comment]
Bump.

This issue isn't specific to the terraform step.  It will impact any project relying on manual intervention.

See the following threads for further discussion (internal links):
- https://octopusdeploy.slack.com/archives/C0GKZRNP4/p1552951721034800
- https://octopusdeploy.slack.com/archives/C033W4273/p1552958178040800

The consensus is the current behavior is incorrect.

The root cause is that we used the `Queued` task state for deployment tasks awaiting manual intervention.  Perhaps we need a new state (e.g. `AwaitingIntervention`).

**Comment by alexrolleyoctopus** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Deployment to an environment should not be allowed if a pending deployment is waiting on manual intervention for the same project](https://trello.com/c/IRhcHp9A/416-deployment-to-an-environment-should-not-be-allowed-if-a-pending-deployment-is-waiting-on-manual-intervention-for-the-same-projec)

**Comment by bryanroth** [comment]
Any updates on this issue?

**Comment by bryanroth** [comment]
Did the fix for this happen to get merged into 2019.5.7?

**Comment by Waldo000000** [comment]
Sorry @bryanroth, it wasn't merged into 2019.5.7. As soon as it's merged, this issue's `Milestone` will be updated with the relevant version.

**Comment by octoreleasebot** [comment]
Release Note: Deployments awaiting manual intervention block other deployments to the same environment

**Comment by bryanroth** [comment]
Thank you!

**Comment by tfbryan** [comment]
@MJRichardson / @Waldo000000  Can you please clarify this change for me? The wording of the title is much different than the wording of the Release Note. 

For example, if I deploy some project to QA and it has a Manual Intervention step, what exactly is now blocked? Are all further deployments to QA blocked?

Thank you,

**Comment by Waldo000000** [comment]
Sorry for the confusion, @tfbryan. If you deploy some project to an environment/tenant and it has a manual intervention step, further deployments _of that project to that environment/tenant_ will be blocked until all such previous deployments have been completed/aborted.

**Comment by MJRichardson** [comment]
Implementing this change caused [problems for existing scenarios](https://github.com/OctopusDeploy/Issues/issues/5666).

Essentially this comes down to two use cases of the manual intervention step:
1.  As an approval mechanism to indicate a release can be promoted.  Typically at the start or end of the process.
2. As a "manual intervention" in the middle of a deployment process.

We felt both these scenarios are valid.  So in `2019.6.3` we implemented the ability to control this on the step: https://github.com/OctopusDeploy/Issues/issues/5666

Note that we reverted the default behaviour to what it was previously. i.e. 
**Manual interventions by default do not block deployments.**

This means that if you wish a manual intervention step to block deployments, you will need to change the option from the default:

![image](https://user-images.githubusercontent.com/113759/60779448-2928c600-a17e-11e9-9e9a-c69ffb5a4a3f.png)
